tests/test_source_yaml.py

```diff
@@ -166,3 +166,65 @@ def settings_customise_sources(
 
     s = Settings()
     assert s.model_dump() == {'yaml3': 3, 'yaml4': 4}
+
+
+@pytest.mark.skipif(yaml is None, reason='pyYAML is not installed')
+def test_yaml_config_section(tmp_path):
+    p = tmp_path / '.env'
+    p.write_text(
+        """
+    foobar: "Hello"
+    nested:
+        nested_field: "world!"
+    """
+    )
+
+    class Settings(BaseSettings):
+        nested_field: str
+
+        model_config = SettingsConfigDict(yaml_file=p, yaml_config_section='nested')
+
+        @classmethod
+        def settings_customise_sources(
+            cls,
+            settings_cls: type[BaseSettings],
+            init_settings: PydanticBaseSettingsSource,
+            env_settings: PydanticBaseSettingsSource,
+            dotenv_settings: PydanticBaseSettingsSource,
+            file_secret_settings: PydanticBaseSettingsSource,
+        ) -> tuple[PydanticBaseSettingsSource, ...]:
+            return (YamlConfigSettingsSource(settings_cls),)
+
+    s = Settings()
+    assert s.nested_field == 'world!'
+
+
+@pytest.mark.skipif(yaml is None, reason='pyYAML is not installed')
+def test_invalid_yaml_config_section(tmp_path):
+    p = tmp_path / '.env'
+    p.write_text(
+        """
+    foobar: "Hello"
+    nested:
+        nested_field: "world!"
+    """
+    )
+
+    class Settings(BaseSettings):
+        nested_field: str
+
+        model_config = SettingsConfigDict(yaml_file=p, yaml_config_section='invalid_key')
+
+        @classmethod
+        def settings_customise_sources(
+            cls,
+            settings_cls: type[BaseSettings],
+            init_settings: PydanticBaseSettingsSource,
+            env_settings: PydanticBaseSettingsSource,
+            dotenv_settings: PydanticBaseSettingsSource,
+            file_secret_settings: PydanticBaseSettingsSource,
+        ) -> tuple[PydanticBaseSettingsSource, ...]:
+            return (YamlConfigSettingsSource(settings_cls),)
+
+    with pytest.raises(KeyError):
```

Comment:
pelase test the error message as well:
```suggestion
    with pytest.raises(KeyError, match='...'):
```

Reply:
Oh, thank you. I added it in [f93364b](https://github.com/pydantic/pydantic-settings/pull/603/commits/f93364ba26df82ef6c4ed182697b5161e24f3132).